apps/api/src/domain/models/term/commands/add-audio-for-term/add-audio-for-term.command-handler.ts

```diff
@@ -0,0 +1,52 @@
+import { AggregateType } from '@coscrad/api-interfaces';
+import { CommandHandler } from '@coscrad/commands';
+import { Valid } from '../../../../../domain/domainModelValidators/Valid';
+import { DeluxeInMemoryStore } from '../../../../../domain/types/DeluxeInMemoryStore';
+import { InMemorySnapshot, ResourceType } from '../../../../../domain/types/ResourceType';
+import { InternalError, isInternalError } from '../../../../../lib/errors/InternalError';
+import { isNotFound } from '../../../../../lib/types/not-found';
+import { ResultOrError } from '../../../../../types/ResultOrError';
+import { BaseUpdateCommandHandler } from '../../../shared/command-handlers/base-update-command-handler';
+import { BaseEvent } from '../../../shared/events/base-event.entity';
+import { EventRecordMetadata } from '../../../shared/events/types/EventRecordMetadata';
+import { Term } from '../../entities/term.entity';
+import { AddAudioForTerm } from './add-audio-for-term.command';
+import { AudioAddedForTerm } from './audio-added-for-term.event';
+
+@CommandHandler(AddAudioForTerm)
+export class AddAudioForTermCommandHandler extends BaseUpdateCommandHandler<Term> {
+    protected actOnInstance(instance: Term, { audioItemId }: AddAudioForTerm): ResultOrError<Term> {
+        return instance.addAudio(audioItemId);
+    }
+
+    protected async fetchRequiredExternalState({
+        audioItemId,
+    }: AddAudioForTerm): Promise<InMemorySnapshot> {
+        const audioItemSearchResult = await this.repositoryProvider
```

Comment:
I'd like to look into how much faster this optimization is. I need to make sure that fetch by ID is actually just fetching by ID in the database layer with an AQL query.